src/layer/vector/Canvas.js

```diff
@@ -219,8 +223,8 @@ L.Canvas = L.Renderer.extend({
 	_handleMouseHover: function (e, point) {
 		var id, layer;
 		if (!this._hoveredLayer) {
-			for (id in this._layers) {
-				layer = this._layers[id];
+			for (id in this._drawnLayers) {
+				layer = this._drawnLayers[id];
```

Comment:
If `drawLayers` is an array, it should be iterated with a usual `for` loop since it's much faster than `for in`. Or you might want to consider @wpf500's suggestion to turn it into an id-keyed object.